grants/teonite/data.md

```diff
@@ -0,0 +1,50 @@
+---
+Grant-Number: "N/A",
+Name: "Smart Contract Source Code Verification Service",
+Service-Provider: "",
+Status: "In progress",
+Grant-Link: "",
+Website: "https://teonite.com/",
+Grant-Size: "Large",
+Total-Milestones: "5",
+Current-Milestone: "M1",
+M1: "11.09.2023","In progress",
+M2: "25.09.2023","To do",
+M3: "02.10.2023","To do",
+M4: "14.08.2023","To do",
+M5: "21.08.2023","To do",
+Twitter: "",
+Public-Project-Chat: "",
+Casper-PM: "https://github.com/piotr-dziubecki",
+Project-GitHub-Repo: "",
+---
+<!--lang:en--> 
+CSPR Live Block Explorer (or in fact any other Casper open-source block explorers) do not implement any services for smart contracts source code and deployment validation or smart contract source code browsing and interacting.
+Our aim is to create: 
+a) source code storage, validation, and browsing service that will expose an API for interacting with any service that would like to incorporate this functionality (CSPR Live or any other block explorer) and will have enterprise-grade security (with Hardware Security Modules - HSM - proposed: Nitro HSM 2)
+b) distributed build system supporting a defined library of Rust compilers and casper-contract smart contract library versions in order to quickly produce WASM builds for further validation
+c) extend casper-client (as a pull request) with functionalities for submitting the source code to the validation service
+d) example web service (and React components library) - that will use the validation service and based on the submitted deploy hash can state if the deployment is validated - if so, enable to browsing the validated deploy source code
+All milestones are in detailed outlined (and calculated in terms of boudges and schedule) based on the following Work Brakedown Structure:
```

Comment:
A couple of typo nitpickings;
- **boudges** instead of **budgets**
- Work **Brakedown** Structure instead of Work **Breakdown** Structure